engines/ags/shared/gfx/allegro_bitmap.cpp

```diff
@@ -123,13 +123,36 @@ void Bitmap::Destroy() {
 	_isDataOwner = false;
 }
 
+BITMAP *Bitmap::ConvertToAllegroFormat(BITMAP *al_bmp) {
+	Graphics::ManagedSurface *origSurface = new Graphics::ManagedSurface(al_bmp->getSurface());
+	Graphics::ManagedSurface *newSurface = nullptr;
+
+	if (origSurface->format.bytesPerPixel == 2 && origSurface->format != Graphics::PixelFormat(2, 5, 6, 5, 0, 11, 5, 0, 0)) { // convert to RGB565
+		newSurface = new Graphics::ManagedSurface(origSurface->w, origSurface->h, Graphics::PixelFormat(2, 5, 6, 5, 0, 11, 5, 0, 0));
+	} else if ((origSurface->format.bytesPerPixel == 3) ||
+			   (origSurface->format.bytesPerPixel == 4 && origSurface->format != Graphics::PixelFormat(4, 8, 8, 8, 8, 16, 8, 0, 24))) { // convert to RGBA8888
+		newSurface = new Graphics::ManagedSurface(origSurface->w, origSurface->h, Graphics::PixelFormat(4, 8, 8, 8, 8, 16, 8, 0, 24));
+	} else { // no conversion needed
+		return al_bmp;
+	}
+
+	if (newSurface) {
+		newSurface->blitFrom(*origSurface);
+		BITMAP *conv_al_bmp = new BITMAP(newSurface);
+		return conv_al_bmp;
```

Comment:
Shouldn't `al_bmp` freed in this case?

Reply:
Thanks, I updated the PR to free the temp surfaces